Doberman/AlarmMonitor.py

```diff
@@ -174,20 +174,23 @@ def check_for_alarms(self):
         doc_filter = {'acknowledged': 0}
         messages = {}
         updates = {'$set': {'acknowledged': dtnow()}}
-        db_col = ('logging', 'alarm_history')
-        if self.db.count(*db_col, doc_filter) == 0:
+        db_col = 'alarm_history'
+        if self.db.count(db_col, doc_filter) == 0:
             return
-        for doc in self.db._check(*db_col).aggregate([
-            {'$match': doc_filter},
-            {'$group': {
-                '_id': '$howbad',
-                'logged': {'$push': {'$toDate': '$_id'}},
-                'msgs': {'$push': '$msg'}}}]):
-            message = '\n'.join([f'{d.iso_format()}: {m}' for d,m in zip(doc['logged'], doc['msgs'])])
-            self.send_message(doc['_id'], message)
+        alarms = {}
+        for doc in self.db.read_from_db(db_col, doc_filter):
+            level = doc['level'] + doc['escalation']
+            logged = datetime.fromtimestamp(int(str(doc['_id'])[:8], 16))
+            if level not in alarms:
+                alarms[level] = {'logged': [], 'msgs': []}
```

Comment:
I'm pretty sure this fails. You can't use a simple int as a dictionary key, but if you tuple it `(level,)` then it works.

Reply:
It does work.

Reply:
I feel like this got patched in a recent python version, I recall this failing in the past. As long as it works now (which it does seem to do)